ctats/counter.go

```diff
@@ -34,17 +19,16 @@ func getOrCreateCounter(
 	id string,
 ) (adder, error) {
 	id = formatID(id)
+	b := fromCtx(ctx)
 
-	b, ctr := counterFromCtx(ctx, id)
-	if ctr != nil {
-		return ctr, nil
-	}
-
-	if b.initializedToNoop {
-		na := &noopAdder{}
-		b.counters[id] = na
+	var ctr adder
 
-		return na, nil
+	if b != nil {
```

Comment:
since we're not erroring out if `b == nil`, would it be sufficient to have these checks in the methods on `b` itself? For example

```go
func (b *bus) getCounter(id string) adder {
  if b == nil {
    return nil
  }

  // existing code for b != nil case
}
```

The same could be done below where we store the counter if the store operation is converted to a method on `b`

Reply:
I'd like some space to rework that to a bit of refinement.  Centralizing around the bus as controller more than a quick data grab.  But I need a little brainspace to make that happen first.